### PR TITLE
Add Docker Builds & Push

### PR DESCRIPTION
**Pull request by JayDwee**
## Summary

This pull request contains all necessary code for automated docker builds and pushes to `jaydwee/koalabot`
Instructions for use are in the README.md

Creation of images has been tested separately, but this will need to be merged before uploaded to docker hub

A config path can now be specified with:
```py
python3 KoalaBot.py --config <path to config>
```
Docstrings and some tests are also added to KoalaDBManager

## Checklist

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

<br>

- [x] Have you tested the changes? ([pytest](https://docs.pytest.org/) & [dpytest](https://dpytest.readthedocs.io/))
- [x] Have you followed [PEP-8](https://www.python.org/dev/peps/pep-0008/) for naming and styling?  
- [x] Has your code been properly documented with RestructuredText docstrings?
- [x] Have you added your changes to `CHANGELOG.md` under the `[Unreleased]` heading?
- [ ] If your code added new bot commands, have you updated `documentation.json`?

<br>

- [x] All of this code is your own, or follows the author repo's licence.

### CODE REVIEW

**Comment by JayDwee** [comment]
⚠️ **DO NOT MERGE YET!** ⚠️

**Review by Kaspiaan — APPROVED** [review]
All seems good, going to request Stefan/Viraj also have a look to make sure the docker stuff is ok

File: .github/workflows/build.yml
Comment (stefan-cooper) [inline .github/workflows/build.yml:40]: do you think there's value in creating a koalabot account on dockerhub and tag the image there instead of using your own personal account?

File: README.md
Comment (stefan-cooper) [inline README.md:97]: is this a typo for `ENCRYPTION`, and what does it do if you enable it?

File: Dockerfile
Comment (stefan-cooper) [inline Dockerfile:21]: any reason not to install these in the install line above?

Comment (JayDwee) [reply Dockerfile:21]: I was having some issues where it didn't install sqlcipher correctly when on the same line, but only on some devices 🤷‍♂️. Separating it seemed to fix the issue

File: README.md
Comment (JayDwee) [reply README.md:97]: Yep, that is a typo, mb. Will fix when I get home.

It overrides whether encryption with sqlcipher is enabled or disabled. Enabled by default

File: .github/workflows/build.yml
Comment (JayDwee) [reply .github/workflows/build.yml:40]: I might look into it in the future, but for now it should be fine, especially if we are the only ones using the docker image

File: README.md
Comment (stefan-cooper) [reply README.md:97]: awesome, i think pending that change then everything else LGTM!

**Review by VirajShah18 — APPROVED** [review]

**Pull request merged**
